docs/fundamentals/runtime-libraries/system-reflection-emit-persistedassemblybuilder.md

```diff
@@ -0,0 +1,126 @@
+
+- If `Reflection.Emit` is used to generate an assembly that will only be executed on the same runtime version as the runtime version that the compiler is running on (typically in-proc), the core assembly can be simply `typeof(object).Assembly`. The following example demonstrates how to create and save an assembly to a stream and run it with the current runtime assembly:
+
+  :::code language="csharp" source="./snippets/System.Reflection.Emit/PersistedAssemblyBuilder/Overview/csharp/CreateAndRunAssembly.cs" id="Snippet1":::
+
+- If `Reflection.Emit` is used to generate an assembly that targets a specific TFM, open the reference assemblies for the given TFM using `MetadataLoadContext` and use the value of the [MetadataLoadContext.CoreAssembly](xref:System.Reflection.MetadataLoadContext.CoreAssembly) property for `coreAssembly`. This value allows the generator to run on one .NET runtime version and target a different .NET runtime version. You should use types returned by the `MetadataLoadContext` instance when referencing core types. For example, instead of `typeof(int)`, find the `System.Int32` type in `MetadataLoadContext.CoreAssembly` by name:
+
+  :::code language="csharp" source="./snippets/System.Reflection.Emit/PersistedAssemblyBuilder/Overview/csharp/CreateAndRunAssembly.cs" id="Snippet2":::
+
+## Set entry point for an executable
+
+To set the entry point for an executable or to set other options for the assembly file, you can call the `public MetadataBuilder GenerateMetadata(out BlobBuilder ilStream, out BlobBuilder mappedFieldData)` method and use the populated metadata to generate the assembly with desired options, for example:
+
+:::code language="csharp" source="./snippets/System.Reflection.Emit/PersistedAssemblyBuilder/Overview/csharp/GenerateMetadataSnippets.cs" id="Snippet1":::
+
+## Emit symbols and generate PDB
+
+The symbols metadata is populated into the `pdbBuilder` out parameter when you call the `GenerateMetadata(out BlobBuilder ilStream, out BlobBuilder mappedFieldData, out MetadataBuilder pdbBuilder)` method on a `PersistedAssemblyBuilder` instance. To create an assembly with a portable PDB:
+
+1. Create `ISymbolDocumentWriter` instances with the `ModuleBuilder.DefineDocument(...)` method. While emitting the method's IL, also emit the corresponding symbol info.
+2. Create a `PortablePdbBuilder` instance using the `pdbBuilder` instance produced by the `GenerateMetadata(...)` method.
+3. Serialize the `PortablePdbBuilder` into a `Blob`, and write the `Blob` into a PDB file stream (only if you're generating a standalone PDB).
+4. Create a `DebugDirectoryBuilder` instance and add a `CodeViewEntry` (standalone PDB) or `EmbeddedPortablePdbEntry`.
```

Comment:
```suggestion
4. Create a <xref:System.Reflection.PortableExecutable.DebugDirectoryBuilder> instance and add a `CodeViewEntry` (standalone PDB) or `EmbeddedPortablePdbEntry`.
```

Reply:
Where are CodeViewEntry and EmbeddedPortablePdbEntry defined?

Reply:
In DebugDirectoryBuilder, I added the link into your comment https://learn.microsoft.com/dotnet/api/system.reflection.portableexecutable.debugdirectorybuilder.addcodeviewentry